packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanBarRow.tsx

```diff
@@ -104,20 +104,22 @@ export default class SpanBarRow extends React.PureComponent<SpanBarRowProps> {
       duration,
       hasChildren: isParent,
       operationName,
-      process: { serviceName, tags },
+      process: { serviceName, tags: processTags },
+      tags: spanTags
     } = span;
     const label = formatDuration(duration);
     const viewBounds = getViewedBounds(span.startTime, span.startTime + span.duration);
     const viewStart = viewBounds.start;
     const viewEnd = viewBounds.end;
 
-    const appSource: KeyValuePair = tags.reduce((acc, val) => val.key == "app.source" ? val : acc);
-    const isFrontEndSpan = appSource.value == "web";
+    const isFrontEndSpan = processTags.some((tag) => tag.key === "app.source" && tag.value === "web");
 
-    var libraryName: string = serviceName;
+    let frontendSpanHeader: string = serviceName;
     if (isFrontEndSpan) {
-      const libraryTag = tags.reduce((acc, val) => val.key == "otel.library.name" ? val : acc);
-      libraryName = libraryTag.value;
+      const otelLibraryName = spanTags.reduce((acc, val) => val.key === "otel.library.name" ? val.value : acc, undefined);
```

Comment:
The `otel.library.name` is in `spanTags` not `processTags` interestingly enough.

Reply:
Ah thank you! That's why it wasn't showing up, great catch!